rotkehlchen/chain/ethereum/names.py

```diff
@@ -65,58 +64,146 @@ def find_ens_mappings(
-        reverse_ens = db_ens.get_reverse_ens(cursor, addresses=addresses).values()
-        mapped_tokens = GlobalDBHandler().get_tokens_mappings(addresses=addresses)
-        hardcoded_mappings = {}
+        settings = database.get_settings(cursor)
+
+    prioritized_addresses = prioritizer.get_prioritized_names(
+        settings.address_name_priority,
+        addresses,
+    )
+
+    return prioritized_addresses
+
+
+FetcherFunc = Callable[[DBHandler, ChecksumEvmAddress], Optional[str]]
+
+
+class NamePrioritizer:
+    _fetchers: Dict[AddressNameSource, FetcherFunc] = {}
+    _db: DBHandler
+
+    def __init__(self, database: DBHandler):
+        self._db = database
+
+    def add_fetchers(self, fetchers: Dict[AddressNameSource, FetcherFunc]) -> None:
+        self._fetchers.update(fetchers)
+
+    def get_prioritized_names(
+            self,
+            prioritized_name_source: List[AddressNameSource],
+            addresses: List[ChecksumEvmAddress],
+    ) -> Dict[ChecksumEvmAddress, Optional[str]]:
+        """
+        Gets the name from the name source with the highest priority.
+        Name source ids with lower index have a higher priority.
+        """
+        top_prio_names: Dict[ChecksumEvmAddress, Optional[str]] = {}
+
         for address in addresses:
-            constant_name = ETHADDRESS_TO_KNOWN_NAME.get(address)
-            if constant_name is not None:
-                hardcoded_mappings[address] = constant_name
-
-        with db_addressbook.read_ctx(AddressbookType.GLOBAL) as global_cursor:
-            global_addressbook = db_addressbook.get_addressbook_entries(
-                cursor=global_cursor,
-                addresses=addresses,
-            )
-        private_addressbook = db_addressbook.get_addressbook_entries(
-            cursor=cursor,
-            addresses=addresses,
-        )
-        addresses_set = set(addresses)
-        labels = {}
-        accounts_data = database.get_blockchain_account_data(
-            cursor=cursor,
-            blockchain=SupportedBlockchain.ETHEREUM,
+            for name_source in prioritized_name_source:
+                fetcher = self._fetchers.get(name_source)
+                if not fetcher:
+                    raise NotImplementedError(
+                        f'address name fetcher for "{name_source}" is not implemented',
+                    )
+
+                name = fetcher(self._db, address)
+                if not name:
+                    continue
+                top_prio_names[address] = name
+                break
+
+        return top_prio_names
+
+
+def blockchain_address_to_name(
+        db: DBHandler,
+        address: ChecksumEvmAddress,
+) -> Optional[str]:
+    """Returns the label of an ethereum blockchain account with the given address or
+    None if there is no such account or the account has no label set.
+    """
+    with db.conn.read_ctx() as cursor:
+        return db.get_blockchain_account_label(
```

Comment:
nitpick for this and the rest. We like kwargs in rotki codebase, since this:

1. won't error due to mistaken argument order
2. python interpreter will actually warn you if the argument does not exist even at parsing time